heat/core/tests/test_arithmetics.py

```diff
@@ -646,6 +646,10 @@ def test_prod(self):
         self.assertEqual(shape_split_axis_tuple_prod.split, None)
         self.assertTrue((shape_split_axis_tuple_prod == expected_result).all())
 
+        # empty array
+        empty = ht.array([])
+        self.assertEqual(ht.prod(empty), ht.array([1.0]))
```

Comment:
the introduced change affects distributed DNDarrays, but tests are on non-distributed arrays.

Reply:
Hi Claudia, from what I understand, the change actually affects non-distributed (to be more specific, empty) DNDarrays. In the linked issue #1017, calling `prod` or `sum` on empty arrays would cause them to (erroneously) enter this if clause. This causes the error in `split + 1` since `split == None`. The introduction of the check to verify that the array is distributed fixes this by skipping the if clause for non-distributed arrays.

Reply:
Got it, thanks!